site/content/entry/citra-progress-report-2018-Q1.md

```diff
@@ -0,0 +1,245 @@
+[here](https://github.com/citra-emu/citra/pull/3711)).
+The old way to upload the data (both vertex and constants) was a simple glBufferData
+call. But keep in mind that the GPU has a huge execution latency, so you can't just
+copy the new content over the existing buffer. So the driver has to choose between
+different ways to deal with this issue, with different drawbacks:
+
+* 1) Wait for the GPU to finish and copy to the destination buffer. Usually the
+fastest for big uploads, fine on common gaming, terrible for emulators.
+* 2) Allocate a new buffer and tell the GPU which new buffer shall be used now.
+You also need to care about freeing the old buffer once it isn't in used any more.
+This tends to a huge CPU overhead and isn't available on partial buffer updates
+(for eg. Lighting LUTs).
+* 3) Copy the data to an internal staging buffer and ask the GPU to copy it again
+to the destination when it is ready. Twice the memory bandwidth, and switching
+between rendering and copy also takes a while for the GPU. This bottlenecks you
+hard if you do this per draw call.
+
+But in the end, all of them are bad in terms of performance. It is the simple
+upload call, but it either requires many copies and context switches, or it stalls
+the GPU.(_More on that in [here](https://de.slideshare.net/CassEveritt/approaching-zero-driver-overhead)_).
+So our buffer streaming way is a bit different. Now, we allocate a big
+staging buffer ourself, copy the data to the current position within the buffer,
+and tell the GPU to directly access the data from there. The driver has no scope
+of interpreting what we ask it to do.
+
+_This has a downside as well_. The GPU now accesses the main memory over PCIe.
+So if we're going to reuse the same data in the next frame, it has to go through
+the PCIe bottleneck again. But, we never use the same data again and hence gain
+a big performance boost.
+
+[wwylele](https://github.com/wwylele) tweaked the configuration UI and renamed
+some parts of it. However, as users became accustomed to the previous UI (while
+it was still in Canary) many of them were confused on where the earlier options went.
+
+#### **What's changed in the UI?**
+
+With the initial release of GLvtx, we've had support for running shaders on the
+host GPU . So, we had a dropdown menu called **Shader Emulation** to select either
+CPU or GPU. The option has now been renamed as **Enable Hardware Shader**. When
+enabled, it makes use of host GPU and when disabled, it uses CPU.
+
+Now as many of the users were using older hardware, we introduced an option called
+**Accurate Hardware Shader**. It was later renamed to **Accurate Multiplication**,
+because technically that's what it was doing. ***When enabled it would render games
+with more accuracy at the cost of slower performance***. Its the same with
+**Accurate Geometry Shader**. Earlier it was configurable via the ini file, but
+now its a check box to enable or disable accurate geometry shader.
+
+&nbsp;
```

Comment:
Is this needed?

Reply:
it leaves some space for the images to appear. after the paragraph.